indexer/services/ender/src/lib/candles-generator.ts

```diff
@@ -432,6 +466,28 @@ export class CandlesGenerator {
       ).toFixed(),
       usdVolume: Big(existingCandle.usdVolume).plus(blockCandleUpdate.usdVolume).toFixed(),
       trades: existingCandle.trades + blockCandleUpdate.trades,
+      orderbookMidPriceClose: existingCandle.orderbookMidPriceClose ?? undefined,
+      orderbookMidPriceOpen: existingCandle.orderbookMidPriceOpen ?? undefined,
+    };
+
+    return CandleTable.update(candle, this.writeOptions) as Promise<CandleFromDatabase>;
+  }
+
+  private async updateCandleWithOrderbookMidPriceInPostgres(
+    existingCandle: CandleFromDatabase,
+    orderbookMidPrice: OrderbookMidPrice,
+  ): Promise<CandleFromDatabase> {
+
+    const candle: CandleUpdateObject = {
+      id: existingCandle.id,
+      low: existingCandle.low,
+      high: existingCandle.high,
+      close: existingCandle.close,
+      baseTokenVolume: existingCandle.baseTokenVolume,
+      usdVolume: existingCandle.usdVolume,
+      trades: existingCandle.trades,
+      orderbookMidPriceOpen: existingCandle.orderbookMidPriceOpen ?? undefined,
```

Comment:
Don't think you need this line. If orderbookMidPriceOpen is not included in the update object, it won't be updated.

Reply:
The `existingCandle` is a `CandleFromDatabase` type which can be null so we need to coerce that into undefined.